### ISSUE TITLE
[4.1.x] DataTable's Export > View Columns does not show all columns if there are few entries in the table

### ISSUE THREAD

**Issue by DanielKimmich** [Laravel-Backpack/CRUD#2649]
# Bug report
Does not show all available fields to list

### What I did
Filter and there are few records to show in the list

### What happened
![image](https://user-images.githubusercontent.com/44840413/78614535-3249df00-7845-11ea-9e9a-9a41bc40aa09.png)

### Suggestion
If possible, allow these two buttons (export, visible column) to be independent and to configure the desired position (top, bottom).

Also be able to set the buttons to show only the icon and description, depending on where you use it (example list, view, show)

In my example duplicate to customize button.
```
@if ($crud->actionIs('show'))
	<a href="{{ url($crud->route.'/'.$entry->getKey().'/edit') }}" class="btn btn-sm btn-link"><i class="la la-edit"></i> {{ trans('backpack::crud.edit') }}</a>

@else
	<a href="{{ url($crud->route.'/'.$entry->getKey().'/edit') }}" class="btn btn-sm btn-link" data-toggle="tooltip" data-placement="bottom" title="{{ trans('backpack::crud.edit') }}"><i class="la la-edit"></i> </a>
@endif	
```

### PHP VERSION:
PHP 7.2.11 (cli) (built: Oct 10 2018 02:04:07) ( ZTS MSVC15 (Visual C++ 2017) x64 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

### LARAVEL VERSION:
v6.18.3@4e48acfaba87f08320a2764d36c3b6a4a4112ccf

### BACKPACK VERSION:
4.1.x-dev@113b18061b0b86b204cfbbb728af0880cbb3c820

**Comment by tabacitu** [comment]
Hi @DanielKimmich ,

Thanks for the bug report but... I don't understand 🤣  Could you please rephrase - maybe I understand a second time? Are you talking about fields/columns? Are you talking about the export functionality or about the normal buttons , inside the table, on each line?

I think I do understand one thing, and I agree - that the export buttons should be more customizable. We plan to rewrite the export ENTIRELY, so that it's a Backpack operation, instead of using the DataTables plugin. Rewriting the export functionality will allow us to:
- put the buttons wherever we want (top/bottom button stack)
- export lines that are on 2nd page, 3rd page of results, etc;
- make it more intuitive - instead of two buttons, we'd just have one "Export" button, that opens a modal and lets you
    - pick the export format
    - pick the columns to show in the export
    - reorder the exported columns

**Comment by pxpm** [comment]
@tabacitu @DanielKimmich 

Could this be a CSS issue, where the dropdown is not shown if table is small (have few or none rows) ?? This looks like it.

Let me know.

**Comment by DanielKimmich** [comment]
I have raised several features and that could be confusing. I'm going to separate it into points:

The reported fix occurs because it has few or no rows.

Make the export buttons a Backpack operation, which opens a modal and allows you to choose the export format select the columns to show in the export (excellent).

On the other hand, there is the "view columns" button independent of export, so that the user can select what they want to see in the list. It would be like a filter, but with columns.

Lastly, I suggested that you can configure the button on the line, show only the icon (minimize space) to show more column data in the list.

Cheers,

**Comment by tabacitu** [comment]
Tested, I can confirm it happens! Thanks for raising this @DanielKimmich !

I agree with all your points. And with @pxpm 's conclusion:
- yes, it happens because of CSS - there's an ```overflow-hidden``` on the table wrapper, which prevents from showing all options in the dropdown; @DanielKimmich you can remove that until a better solution is adopted; but as far as I know this makes responsive tables quirky (they don't break, but... they're ugly sometimes); so it's not a long-term solution;
- I agree, I've wanted to work on an "Export" operation for a long long time - I've just taken the time to write it down here https://github.com/Laravel-Backpack/CRUD/issues/2667 ; let's consider this issue about the CSS problem, and talk there about the Export operation;

**Comment by pxpm** [comment]
This issue is beeing addressed here: #2945 

Let's close this one in favor.
